src/node_sqlite.cc

```diff
@@ -664,6 +664,66 @@ void UserDefinedFunction::xDestroy(void* self) {
   delete static_cast<UserDefinedFunction*>(self);
 }
 
+AuthorizerFunction::AuthorizerFunction(Environment* env,
+                                       Local<Function> fn,
+                                       DatabaseSync* db)
+    : env_(env), fn_(env->isolate(), fn), db_(db) {}
+
+AuthorizerFunction::~AuthorizerFunction() {}
+
+int AuthorizerFunction::xAuthorizer(void* user_data,
+                                    int action_code,
+                                    const char* param1,
+                                    const char* param2,
+                                    const char* param3,
+                                    const char* param4) {
+  AuthorizerFunction* self = static_cast<AuthorizerFunction*>(user_data);
+  Environment* env = self->env_;
+  Isolate* isolate = env->isolate();
+  HandleScope handle_scope(isolate);
+  Local<Context> context = env->context();
+
+  auto fn = self->fn_.Get(isolate);
+  LocalVector<Value> js_argv(isolate);
+
+  // Convert SQLite authorizer parameters to JavaScript values
+  js_argv.emplace_back(Integer::New(isolate, action_code));
+  js_argv.emplace_back(
+      NullableSQLiteStringToValue(isolate, param1).ToLocalChecked());
+  js_argv.emplace_back(
+      NullableSQLiteStringToValue(isolate, param2).ToLocalChecked());
+  js_argv.emplace_back(
+      NullableSQLiteStringToValue(isolate, param3).ToLocalChecked());
+  js_argv.emplace_back(
+      NullableSQLiteStringToValue(isolate, param4).ToLocalChecked());
+
+  TryCatch try_catch(isolate);
+  MaybeLocal<Value> retval =
+      fn->Call(context, Undefined(isolate), js_argv.size(), js_argv.data());
+
+  if (try_catch.HasCaught()) {
+    // If there's an exception in the callback, deny the operation
+    return SQLITE_DENY;
+  }
```

Comment:
What if I let it crash? Other than that, a broken function will have its error swallowed making it hard to debug.

Other implementations follow [that pattern](https://github.com/nodejs/node/blob/main/src/node_sqlite.cc#L385-L389)

Reply:
A function might throw an exception for many reasons, we could add an option to return deny when an exception is thrown.

Reply:
Was this comment addressed?

Reply:
Oh, it was. It throws. Pretty good.